target/consensource
# Add Modular OAuth

- Add CORS fairing and CORS routes to handle pre-flight options requests
- Add health check endpoint
- Add support for OAuth/OpenID Connect
- Add request guard on user creation endpoint for web token
- Reduce cost on bcrypt hashing
- Add ID provider integration note to README.md
- Note: Specify `OAUTH_VALIDATION_URL` as environment variable in compose file

## Proposed change/fix

Adds the option to integrate the REST API with an external OAuth/OpenID Connect service. Also adds CORS option endpoints and a health check endpoint.

## Types of changes


## How to run/test
Rebuild `rest-api` and confirm it works as normal.
`docker-compose up --build rest-api`
then
`docker-compose up`

To make sure it fails when specified,
add `OAUTH_VALIDATION_URL: "https://domain.tld"` in the `docker-compose.yaml`

```
  rest-api:
    image: target/consensource:rest-api
    container_name: consensource-rest-api
    build:
      context: .
      dockerfile: ./rest_api/Dockerfile
    ports:
      - "9009:9009"
      - "9010:9010"
    environment:
      ROCKET_ADDRESS: "0.0.0.0"
      ROCKET_PORT: "9009"
      OAUTH_VALIDATION_URL: "https://domain.tld"
    command: |
      bash -c "
        cert-registry-rest-api -v --dbhost postgres -C tcp://validator:4004
      "
```
Then rebuild and restart.
